dev/ProgressBar/InteractionTests/ProgressBarTests.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Common;
+using Windows.UI.Xaml.Tests.MUXControls.InteractionTests.Infra;
+using Windows.UI.Xaml.Tests.MUXControls.InteractionTests.Common;
+using System.Collections.Generic;
+
+#if USING_TAEF
+using WEX.TestExecution;
+using WEX.TestExecution.Markup;
+using WEX.Logging.Interop;
+#else
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.VisualStudio.TestTools.UnitTesting.Logging;
+#endif
+
+using Microsoft.Windows.Apps.Test.Automation;
+using Microsoft.Windows.Apps.Test.Foundation;
+using Microsoft.Windows.Apps.Test.Foundation.Controls;
+using Microsoft.Windows.Apps.Test.Foundation.Patterns;
+using Microsoft.Windows.Apps.Test.Foundation.Waiters;
+
+namespace Windows.UI.Xaml.Tests.MUXControls.InteractionTests
+{
+    [TestClass]
+    public class ProgressBarTests
+    {
+        [ClassInitialize]
+        [TestProperty("RunAs", "User")]
+        [TestProperty("Classification", "Integration")]
+        [TestProperty("Platform", "Any")]
+        public static void ClassInitialize(TestContext testContext)
+        {
+            TestEnvironment.Initialize(testContext);
+        }
+
+        public void TestCleanup()
+        {
+            TestCleanupHelper.Cleanup();
+        }
+
+        [TestMethod]
+        public void ChangeValueTest()
+        {
+            using (var setup = new TestSetupHelper("ProgressBar Tests"))
+            {
+                Log.Comment("Changing Value of ProgressBar");
+
+                UIObject testProgressBar = FindElement.ByName("TestProgressBar");
+                TextBlock valueText = FindElement.ByName<TextBlock>("ValueText");
+
+                double oldValue = Convert.ToDouble(valueText.DocumentText);
+
+                // NOTE: Interaction tests can only access what accessibility tools see. In this case, we can find the button because we
+                // set AutomationProperties.Name on the button in ProgressBarPage.xaml
+                Button changeValueButton = FindElement.ByName<Button>("ChangeValueButton");
+                changeValueButton.InvokeAndWait();
+
+                double newValue = Convert.ToDouble(valueText.DocumentText);
+                double diff = Math.Abs(oldValue - newValue);
+
+                Log.Comment("ProgressBar value changed");
+                Verify.IsGreaterThan(diff, Convert.ToDouble(0));
+            }
+        }
+
+        [TestMethod]
+        public void UpdateMinMaxTest()
```

Comment:
>UpdateMinMaxTest [](start = 20, length = 16)

Have you looked into the test hooks pattern some of the other controls use? I think it would help you write tests which check to make sure the indicator's size is correct and stuff like that.